benthomasson/eda/tests/integration/utils.py

```diff
@@ -0,0 +1,72 @@
+import os
+import subprocess
+from dataclasses import dataclass
+from typing import List
+from typing import Optional
+from time import sleep
+
+from . import TESTS_PATH
+
+
+@dataclass
+class CLIRunner:
+    """
+    Wrapper of subprocess.run to compose cmd's for ansible-events CLI
+    """
+
+    cwd: str = TESTS_PATH
+    base_cmd: str = "ansible-events"
+    inventory: str = os.path.join(TESTS_PATH, "default_inventory.yml")
+    rules: Optional[str] = None
+    sources: Optional[str] = None
+    extra_vars: Optional[str] = None
+    envvars: Optional[str] = None
+    proc_id: Optional[str] = None
+    verbose: bool = False
+    debug: bool = False
+    timeout: float = 10.0
+
+    def _process_args(self) -> List[str]:
+        args = [
+            self.base_cmd,
+        ]
+
+        args.extend(["-i", self.inventory])
+
+        if self.rules:
+            args.extend(["--rules", self.rules])
+        if self.sources:
+            args.extend(["-S", self.sources])
+        if self.extra_vars:
+            args.extend(["--vars", self.extra_vars])
+        if self.envvars:
+            args.extend(["--env-vars", self.envvars])
+        if self.proc_id:
+            args.extend(["--id", self.proc_id])
+        if self.verbose:
+            args.append("--verbose")
+        if self.debug:
+            args.append("--debug")
+
+        return args
+
+    def run(self):
+        args = self._process_args()
+        return subprocess.run(
+            args, cwd=self.cwd, capture_output=True, timeout=self.timeout, check=True
+        )
+
+    def run_in_background(self):
+        args = self._process_args()
+        proc = subprocess.Popen(
+            args, cwd=self.cwd, stdout=subprocess.DEVNULL, stderr=subprocess.PIPE
+        )
+
+        # Ensure ansible-events is running
+        for i in range(10):
```

Comment:
We require this logic to ensure ansible-events starts up and is "ready", otherwise we hit a race condition where the client tries to post data before the webhook server is ready. If we move this logic to the caller then we would just repeat the logic in every other source plugin that will use this method, which seems pointless imo.

Reply:
What if you run a process that you expect to fail? If you don't want to repeat code you can encapsulate it, but that responsibility clearly does not belong to this method.

Reply:
I have moved the logic to the caller as suggested.